aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -470,30 +521,38 @@ module AptosFramework::Stake {
         pool_address: address,
         owner_cap: &OwnerCapability,
         coins: Coin<TestCoin>,
-    ) acquires StakePool, StakePoolEvents, ValidatorSet, ValidatorSetConfiguration {
+    ) acquires StakePool, StakePoolEvents, ValidatorSet, ValidatorSetConfiguration, ValidatorSetVotingPower {
         assert!(owner_cap.pool_address == pool_address, Errors::invalid_argument(ENOT_OWNER));
+        let amount_added = Coin::value<TestCoin>(&coins);
 
+        let validator_set_config = borrow_global<ValidatorSetConfiguration>(@CoreResources);
         let stake_pool = borrow_global_mut<StakePool>(pool_address);
-        let amount = Coin::value<TestCoin>(&coins);
         // Add to pending_active if it's a current validator because the stake is not counted until the next epoch.
         // Otherwise, the delegation can be added to active directly as the validator is also activated in the epoch.
         if (is_current_validator(pool_address)) {
             Coin::merge<TestCoin>(&mut stake_pool.pending_active, coins);
+
+            // Update the amount of stake being added to the active validator set's total voting power.
```

Comment:
I think there's still some nuance here, is_current_validator includes the pending_inactive nodes too.

if the node is in the pending_active queue (not current validator), we should count the added value as well

Reply:
Updated